src/main/scala/org/broadinstitute/dsde/workbench/sam/Boot.scala

```diff
@@ -133,7 +136,23 @@ object Boot extends IOApp with LazyLogging {
         backgroundAccessPolicyDAO,
         backgroundLdapExecutionContext,
         blockingEc)
-    } yield createAppDepenciesWithSamRoutes(appConfig, cloudExtensionsInitializer, foregroundAccessPolicyDAO, foregroundDirectoryDAO, registrationDAO, identityConcentrator)
+
+      userService = new UserService(foregroundDirectoryDAO, cloudExtensionsInitializer.cloudExtensions, registrationDAO, appConfig.blockedEmailDomains)
+
+      cryptominingSubscriberProcess <- appConfig.googleConfig match {
+        case Some(value) => initCryptominingSubscriber(value.googleServicesConfig.cryptominingSubscriber, userService)
+        case None => effect.Resource.liftF[IO, List[fs2.Stream[IO, Unit]]](IO(List.empty))
+      }
+    } yield createAppDepenciesWithSamRoutes(appConfig, cloudExtensionsInitializer, userService, cryptominingSubscriberProcess, foregroundAccessPolicyDAO, foregroundDirectoryDAO, identityConcentrator)
+
+  private def initCryptominingSubscriber(cryptominingSubscriberConfig: SubscriberConfig, userService: UserService)
+                                        (implicit logger: StructuredLogger[IO]): effect.Resource[IO, List[fs2.Stream[IO, Unit]]] = for {
+    subscriberQueue <- cats.effect.Resource.liftF(InspectableQueue.bounded[IO, Event[CryptominingUserMessage]](200))
+    cryptominingSubscriber <- GoogleSubscriber.resource[IO, CryptominingUserMessage](cryptominingSubscriberConfig, subscriberQueue)
+  } yield {
+    val subscriber = new CryptominingSubscriber(cryptominingSubscriber, userService)
+    List(subscriber.process, fs2.Stream.eval(cryptominingSubscriber.start))
```

Comment:
can you explain why this is a list of streams?

Reply:
`cryptominingSubscriber.start` is for starting up the underlying google subscriber, `subscriber.process` is for actually running the stream that processes messages that the subscriber receives